kaldb/src/main/java/com/slack/kaldb/server/ArmeriaService.java

```diff
@@ -38,130 +36,106 @@
+                  RequestContextCurrentTraceContext.builder()
+                      .addScopeDecorator(ThreadContextScopeDecorator.get())
+                      .build())
+              .addSpanHandler(spanHandler)
+              .build();
+      this.serverBuilder.decorator(BraveService.newDecorator(tracing));
     }
 
-    // always add a tracer, even if we're not reporting
-    Tracing tracing =
-        Tracing.newBuilder()
-            .localServiceName(this.serviceName)
-            .currentTraceContext(
-                RequestContextCurrentTraceContext.builder()
-                    .addScopeDecorator(ThreadContextScopeDecorator.get())
-                    .build())
-            .addSpanHandler(spanHandler)
-            .build();
-    sb.decorator(BraveService.newDecorator(tracing));
-  }
+    public Builder withElasticsearchApi(KaldbServiceGrpc.KaldbServiceImplBase searcher) {
+      this.serverBuilder.annotatedService(new ElasticsearchApiService(searcher));
+      return this;
+    }
 
-  private void addManagementEndpoints(ServerBuilder sb) {
-    sb.decorator(
-        MetricCollectingService.newDecorator(GrpcMeterIdPrefixFunction.of("grpc.service")));
-    sb.decorator(getLoggingServiceBuilder().newDecorator());
-    sb.http(serverPort);
-    sb.service("/health", HealthCheckService.builder().build());
-    sb.service("/metrics", (ctx, req) -> HttpResponse.of(prometheusMeterRegistry.scrape()));
-    sb.serviceUnder("/docs", new DocService());
-    sb.serviceUnder("/internal/management/", ManagementService.of());
-  }
+    public Builder withGrpcSearchApi(KaldbServiceGrpc.KaldbServiceImplBase searcher) {
+      GrpcServiceBuilder searchBuilder =
+          GrpcService.builder()
+              .addService(searcher)
+              .enableUnframedRequests(true)
+              .useBlockingTaskExecutor(true);
+      this.serverBuilder.decorator(
+          MetricCollectingService.newDecorator(GrpcMeterIdPrefixFunction.of("grpc.service")));
+      this.serverBuilder.service(searchBuilder.build());
+      return this;
+    }
 
-  private LoggingServiceBuilder getLoggingServiceBuilder() {
-    return LoggingService.builder()
-        // Not logging any successful response, say prom scraping /metrics every 30 seconds at INFO
-        .successfulResponseLogLevel(LogLevel.DEBUG)
-        .failureResponseLogLevel(LogLevel.ERROR)
-        // Remove the content to prevent blowing up the logs
-        .responseContentSanitizer((ctx, content) -> "truncated")
-        // Remove all headers to be sure we aren't leaking any auth/cookie info
-        .requestHeadersSanitizer((ctx, headers) -> DefaultHttpHeaders.EMPTY_HEADERS);
+    public ArmeriaService build() {
+      return new ArmeriaService(serverBuilder.build(), serviceName);
+    }
   }
 
   @Override
   protected void startUp() throws Exception {
-    LOG.info("Starting {} on port {}", serviceName, serverPort);
+    LOG.info("Starting {} on ports {}", serviceName, server.config().ports());
```

Comment:
If there are multiple ports, it's better to list each port number separately here.

Reply:
This is actually intentional - the `ports()` method returns a list of `ServerPort` which has a custom defined `toString()`, and includes a log more information (local address, port, protocols).

```
Starting kalDbIndex on ports [ServerPort(0.0.0.0/0.0.0.0:8080, [http])]
```